users/tests/test_views.py

```diff
@@ -496,3 +496,56 @@ def test_menu_expansion_update_should_update_menu_expansion_setting_in_user_pref
 
         user_menu_expansion_setting_after_update = CustomUserPreferences.objects.get(user=self.employee).expanded_menu
         self.assertEqual(user_menu_expansion_setting_after_update, not expanded_menu)
+
+
+class ChangePasswordOnLoginTests(TestCase):
+    def setUp(self):
+        self.user_password = "password"
+        self.user = UserFactory(need_change_password=True)
+
+    def test_change_user_password_on_login_view_should_change_user_password_on_post(self):
+        data = {"old_password": self.user_password, "new_password1": "newuserpasswd", "new_password2": "newuserpasswd"}
+        self.client.login(email=self.user.email, password=self.user_password)
+        response = self.client.post(reverse("password_change_on_login"), data)
+        self.user.refresh_from_db()
+        self.assertEqual(response.status_code, 302)
```

Comment:
When you add `follow=True` property to `self.client.post` method you will be able to check if you have right `200` status code after this request

Reply:
Ok.